src/Integration.Vsix/Analysis/SingleFileMonitor.cs

```diff
@@ -119,6 +119,26 @@ private static void EnsureDirectoryExists(string filePath, IFileSystem fileSyste
             }
         }
 
+        private bool IsDuplicateChangeEvent(string path)
+        {
+            // We're trying to ignore duplicate events by checking the last-write time.
+            // However, the precision of DateTime means that it is possible for separate events
+            // that happen very close together to report the same last-write time. If that happens,
+            // we'll be ignoring a "real" notification.
+            var currentTime = File.GetLastWriteTimeUtc(path);
+            var timeDiff = currentTime - lastWriteTime;
+
+            if (timeDiff.TotalMilliseconds >= 0 && timeDiff.TotalMilliseconds < 10)
```

Comment:
This is a bit arbitrary indeed. Does it matter if it's 10 milliseconds or 20 or 200? If the file system indeed aggregates the events, then it might take longer and it would still be ok.

In fact, I'm not 100% convinced this class should be implemented at all. It should be up to the caller to handle duplicate events and throttling. In fact, the comment at the top of the class says "we're removing duplicates but at the cost of potentially losing a few real events", which IMO is way worse. 

@duncanp-sonar I vaguely remember us having a discussion about this when we introduced `analysisConfigMonitor.ConfigChanged` for ESLint bridge. I'll try to find out if it was written somewhere.

Reply:
Edit: I believe we did have this discussion when implementing `SuppressedIssuesFileWatcher` and looks like we decided not to use this class. However I couldn't find anything written, I guess we talked about it F2F...
https://github.com/SonarSource/sonarlint-visualstudio/pull/2905